apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/model/BundleComponent.java

```diff
@@ -698,30 +696,21 @@ protected IApiTypeContainer createApiTypeContainer(String path) throws CoreExcep
 						zip = new ZipFile(fLocation);
 						ZipEntry entry = zip.getEntry(path);
 						if (entry != null) {
-							File tmpfolder = new File(System.getProperty("java.io.tmpdir")); //$NON-NLS-1$
 							if (entry.isDirectory()) {
 								// extract the dir and all children
-								File dir = Util.createTempFile(TMP_API_FILE_PREFIX, TMP_API_FILE_POSTFIX);
-								// hack to create a temp directory
-								// see
-								// http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=4735419
-								if (dir.delete()) {
-									dir.mkdir();
-									FileManager.getManager().recordTempFileRoot(dir.getCanonicalPath());
-								}
+								File dir = Util.createTempDirectory(TMP_API_FILE_PREFIX);
 								extractDirectory(zip, entry.getName(), dir);
-								if (dir.isDirectory() && dir.exists()) {
-									return new DirectoryApiTypeContainer(this, dir.getCanonicalPath());
-								}
+								return new DirectoryApiTypeContainer(this, dir.getCanonicalPath());
 							} else {
-								File file = extractEntry(zip, entry, tmpfolder);
-								if (Util.isArchive(file.getName())) {
-									File parent = file.getParentFile();
-									if (!parent.equals(tmpfolder)) {
-										FileManager.getManager().recordTempFileRoot(parent.getCanonicalPath());
-									} else {
-										FileManager.getManager().recordTempFileRoot(file.getCanonicalPath());
-									}
+								// also create a temp directory with unique name first,
+								// then extract the file inside that directory.
+								// Without that additional parent directory, this would clash, e.g.
+								// due to pre-existing files in the temp dir or when comparing with
+								// a baseline during API analysis: one file would simply overwrite
+								// the other one, never finding any issues.
```

Comment:
```suggestion
								// Create a uniquely named temp-directory and extract the file inside it.
								// Without that additional parent directory, this could clash, e.g.
								// due to pre-existing files in the temp dir or when comparing with
								// a baseline during API analysis: one file would simply overwrite
								// the other one and thus a difference cannot be found.
```